File: SurvivorBot v1/src/main/java/com/fireblazerrr/survivorbot/spigot/command/Command.java
package com.fireblazerrr.survivorbot.spigot.command;

import org.bukkit.command.CommandSender;

public interface Command {

    void cancelInteraction(CommandSender cs);

    boolean execute(CommandSender sender, String identifier, String[] args);

    String getDescription();

    String[] getIdentifiers();

    int getMaxArguments();

    int getMinArguments();

    String getName();

    String[] getNotes();

    String getPermission();

    String getUsage();

    boolean isIdentifier(CommandSender cs, String s);

    boolean isInProgress(CommandSender cs);

    boolean inInteractive();

    boolean isShownOnHelpMenu();
}

File: src/main/kotlin/com/fireblazerrr/survivorbot/spigot/command/Command.kt
package com.fireblazerrr.survivorbot.spigot.command

import org.bukkit.command.CommandSender

interface Command{
    fun execute(sender: CommandSender, identifier: String, args: Array<String>): Boolean
    fun isIdentifier(key: String): Boolean
    fun cancelInteraction(commandSender: CommandSender)
    fun isInProgress(commandSender: CommandSender): Boolean
    fun isInteractive(): Boolean
    fun isShownOnHelpMenu(): Boolean

}